src/aics-image-viewer/state/store.ts

```diff
@@ -43,8 +43,27 @@ const VIEWER_SETTINGS_CHANGE_HANDLERS: ViewerSettingChangeHandlers = {
   }),
 };
 
+const validateStateValue = <K extends keyof ViewerState>(
+  currentState: ViewerState,
+  key: K,
+  value: Partial<ViewerState[K]>
+): Partial<ViewerState> => {
+  const changeHandler = VIEWER_SETTINGS_CHANGE_HANDLERS[key];
+
+  if (changeHandler) {
+    // some settings have custom change handlers to avoid creating illegal states; if this one has one, call it
+    return changeHandler(value, currentState);
+  } else {
+    // if not, merge the new value to the current one (if applicable) and return
+    const currentValue = currentState[key];
+    const nextValue = isRecord(currentValue) && isRecord(value) ? { ...currentValue, ...value } : value;
```

Comment:
Is it possible for `currentValue` to not be a record while `value` is? Or is the check just for satisfying type checking?

Reply:
Probably mostly for type checking